src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/latest/test_graph.py

```diff
@@ -366,9 +370,17 @@ def test_graph_app_cred_e2e(self):
             # ensure we can update SP's properties #5948
             self.cmd('az ad sp update --id {app} --set appRoleAssignmentRequired=true')
             self.cmd('az ad sp show --id {app}')
+
+            result = self.cmd('ad sp create-for-rbac --name {app2} --skip-assignment --years 10').get_output_in_json()
+            app_id2 = result['appId']
+            result = self.cmd('ad sp credential list --id {app2}', checks=self.check('length([*])', 1)).get_output_in_json()
+            diff = dateutil.parser.parse(result[0]['endDate']).replace(tzinfo=None) - datetime.datetime.utcnow()
+            self.assertTrue(diff.days > 9 * 365)
```

Comment:
NIT: The verification can be more accurate. Such as diff.days > 9.9 * 365.
We can also verify that diff.days < 10.1 * 365